src/Disks/IDisk.h

```diff
@@ -248,6 +248,10 @@ class IDisk : public Space
     /// Overrode in remote fs disks.
     virtual bool supportZeroCopyReplication() const = 0;
 
+    /// Whether this disk support parallel write
+    /// Overrode in remote fs disks.
+    virtual bool supportParallelWrite() const { return false; }
```

Comment:
btw looks like this never worked because remote disks are wrapped into `DiskDecorator` for `DiskRestartProxy` and `DiskCacheWrapper` (old cache version which is stilled turned on by default) and for `DiskDecorator` this method was not overriden: https://github.com/ClickHouse/ClickHouse/pull/38792

Reply:
@azat the test, added in this PR, does not work after the issue (which I described above) is fixed (see https://github.com/ClickHouse/ClickHouse/pull/38792 checks). If this test does not actually check anything now, I will remove it for now, agree?

Reply:
The test uses explicit `max_insert_delayed_streams_for_parallel_write` so it works correct.

As for the failure in the #38792 you can set `max_insert_delayed_streams_for_parallel_write=0` explicitly for INSERT w/o this option and then it will ignore remote disk or not.

Also I would recommend add another test that will ensure that S3 does uses parallel writes (via `MEMORY_LIMIT_EXCEEDED` during `INSERT` or `thread_ids` from `query_log`)